scripts/joinmarket-qt.py

```diff
@@ -1464,71 +1465,90 @@ def openQRCodePopup(self, address):
         popup.show()
 
     def updateWalletInfo(self, walletinfo=None):
-        nm = jm_single().config.getint("GUI", "max_mix_depth")
-        l = self.walletTree
+        max_mixdepth_count = jm_single().config.getint("GUI", "max_mix_depth")
 
         # before deleting, note whether items were expanded
-        esrs = []
-        for i in range(l.topLevelItemCount()):
-            tli = l.invisibleRootItem().child(i)
-            # must check top and also the two subitems (branches):
-            expandedness = tuple(
-                x.isExpanded() for x in [tli, tli.child(0), tli.child(1)])
-            esrs.append(expandedness)
-        l.clear()
-        if walletinfo:
-            rows, mbalances, xpubs, total_bal = walletinfo
-            if jm_single().config.get("BLOCKCHAIN", "blockchain_source") == "regtest":
-                self.wallet_name = mainWindow.testwalletname
+        previous_expand_states = []
+        for i in range(self.walletTree.topLevelItemCount()):
+            tli = self.walletTree.invisibleRootItem().child(i)
+            # expandedness is a list beginning with the top level expand state,
+            # followed by the expand state of its children
+            expandedness = [tli.isExpanded()]
+            for j in range(tli.childCount()):
+                expandedness.append(tli.child(j).isExpanded())
+            previous_expand_states.append(expandedness)
+        self.walletTree.clear()
+
+        # Skip the remaining of this method if wallet info doesn't exist
+        if not walletinfo:
+            return
+
+        rows, mbalances, xpubs, total_bal = walletinfo
+        if jm_single().config.get("BLOCKCHAIN", "blockchain_source") == "regtest":
```

Comment:
We need to support signet (and technically, "testnet" as in testnet3, though I consider that basically deprecated now).